src/Toolbar.jsx

```diff
@@ -1,33 +1,78 @@
 // @flow
+
 import { Dropdown, Image, Menu } from 'semantic-ui-react';
 import React from 'react';
-import slack from './images/Slack_Icon.png';
-import smile from './images/smile.jpg';
+import slack from './images/slack_icon.png';
+import frustrated from './images/emojis/frustrated.jpg';
+import sad from './images/emojis/sad.jpg';
+import neutral from './images/emojis/neutral.jpg';
+import happy from './images/emojis/happy.jpg';
+import smile from './images/emojis/smile.jpg';
+import convertScoreToColorAndEmoji from './convertScoreToColorAndEmoji';
+
 import './index.css';
+import store from './store';
+import Actions from './Actions';
 
-type Props = {
-  color: string,
-}
+const sentiments =
+  {
+    frustrated,
+    sad,
+    neutral,
+    smile,
+    happy,
+  };
+
+
+export default class Toolbar extends React.Component {
+  componentWillMount() {
+    const channels = Object.keys(store.getState().channelData);
+    if (channels.length === 0) {
+      store.dispatch(Actions.fetchChannels());
```

Comment:
all dispatches to the store need to be done through react-redux's connect function. The only place you should ever need to be importing the store is in src/index.js. nowhere else